Firestore/Source/API/FIRQuery.m

```diff
@@ -256,6 +256,82 @@ - (FIRQuery *)queryWhereFieldPath:(FIRFieldPath *)path isGreaterThanOrEqualTo:(i
                                  value:value];
 }
 
+- (FIRQuery *)queryWherePredicate:(NSPredicate *)predicate {
+  if ([predicate isKindOfClass:[NSComparisonPredicate class]]) {
+    // Comparison predicate
+    NSComparisonPredicate *comparison = (NSComparisonPredicate *)predicate;
+    if (comparison.comparisonPredicateModifier != NSDirectPredicateModifier) {
+      FSTThrowInvalidArgument(@"Invalid query. Predicate cannot have an "
+                              "aggregate modifier.");
+    }
+    NSString *path;
+    id value = nil;
+    if (comparison.leftExpression.keyPath == nil &&
+        comparison.rightExpression.keyPath == nil) {
+      FSTThrowInvalidArgument(@"Invalid query. Predicate comparison must have "
+                              "a key path on either side of its expression.");
+    } else if (comparison.leftExpression.keyPath == nil) {
```

Comment:
nit: This could be clearer if you tested that the key path was non-nil then used it in the immediately following block.

As it stands you're comparing against `leftExpression.keyPath` then using the `rightExpression.keyPath` right below.